src/planscape/datasets/models.py

```diff
@@ -0,0 +1,163 @@
+from django.contrib.auth import get_user_model
+from django.core.validators import URLValidator
+from django.conf import settings
+from django.contrib.gis.db import models
+from django_stubs_ext.db.models import TypedModelMeta
+from treebeard.mp_tree import MP_Node
+
+from core.models import CreatedAtMixin, DeletedAtMixin, UpdatedAtMixin
+from core.schemes import SUPPORTED_SCHEMES
+
+User = get_user_model()
+
+
+class VisibilityOptions(models.TextChoices):
+    PRIVATE = "PRIVATE", "Private"
+    PUBLIC = "PUBLIC", "Public"
+
+
+class Dataset(CreatedAtMixin, UpdatedAtMixin, DeletedAtMixin, models.Model):
+    id: int
+
+    created_by_id: int
+    created_by = models.ForeignKey(
+        User,
+        related_name="created_datasets",
+        on_delete=models.RESTRICT,
+    )
+
+    name = models.CharField(max_length=128)
+    description = models.TextField(
+        null=True,
+    )
+    visibility = models.CharField(
+        choices=VisibilityOptions.choices,
+        default=VisibilityOptions.PRIVATE,
+    )
+    version = models.CharField(
+        null=True,
+    )
+
+    class Meta(TypedModelMeta):
+        verbose_name = "Dataset"
+        verbose_name_plural = "Datasets"
+
+
+class Category(CreatedAtMixin, UpdatedAtMixin, DeletedAtMixin, MP_Node):
+    id: int
+
+    created_by_id: int
+    created_by = models.ForeignKey(
+        User,
+        related_name="created_datasets",
+        on_delete=models.RESTRICT,
+    )
+
+    dataset_id: int
+    dataset = models.ForeignKey(
+        Dataset,
+        related_name="categories",
+        on_delete=models.RESTRICT,
+    )
+
+    order = models.IntegerField(
+        default=0,
+        help_text="If necessary, changing the order allows the users to configure what categories appears first.",
+    )
+
+    node_order_by = [
+        "order",
+        "name",
+    ]
+
+    name = models.CharField(
+        max_length=128,
+    )
+
+    class Meta(TypedModelMeta):
+        verbose_name = "Category"
+        verbose_name_plural = "Categories"
+
+
+class DataLayerType(models.TextChoices):
+    VECTOR = "VECTOR", "Vector"
+    RASTER = "RASTER", "Raster"
+
+
+class DataLayerStatus(models.TextChoices):
+    READY = "READY", "Ready"
+    PENDING = "PENDING", "Pending"
+    FAILED = "FAILED", "Failed"
+
+
+class GeometryType(models.TextChoices):
+    NO_GEOM = "NO_GEOM", "No Geometry Field"
+    RASTER = "RASTER", "Raster"
+    POINT = "POINT", "Point"
+    MULTIPOINT = "MULTIPOINT", "MultiPoint"
+    LINESTRING = "LINESTRING", "LineString"
+    MULTILINESTRING = "MULTILINESTRING", "MultiLineString"
+    POLYGON = "POLYGON", "MultiPolygon"
```

Comment:
Should this be `POLYGON = "POLYGON", "Polygon"` ?

Also- should we have this distintion (poly vs multi ) or is this something we should simplify and transform so we only deal with one type? 


Same with `linestring / MultiLineString` (i dont know what is the difference so not sure that makes sense)

Reply:
@george-silva still seeing POLYGON = "POLYGON", "MultiPolygon" here and on the migrations